test/no_invalidations.jl

```diff
@@ -0,0 +1,110 @@
+                    @eval $expr
+                    invs
+                end
+            finally
+                ccall(:jl_debug_method_invalidation, Any, (Cint,), 0)
+            end
+        end
+    end
+    function there_are_no_invalidations_expr(expr::Expr)
+        e = invalidations_expr(expr)
+        quote
+            let invalidations = $e
+                show(isempty(invalidations))
+            end
+        end
+    end
+    function there_are_no_invalidations(expr::Expr)
+        e = there_are_no_invalidations_expr(expr)
+        s = readline_julia_expr_cmd(e)
+        parse(Bool, s)
+    end
+end
+
+function type_expr(f, supertype::Type)
+    type_name = :T
+    rest = f(type_name)
+    quote
+        struct $type_name <: $supertype end
+        $rest
+    end
+end
+
+function test_expr(f, supertype::Type)
+    e = type_expr(f, supertype)
+    ThereAreNoInvalidations.there_are_no_invalidations(e)
+end
+
+@testset "no invalidations test set" begin
+    @testset "2-arg `show` for various new types" begin
+        stypes = (Integer, AbstractString, DenseVector{UInt8}, DenseMatrix{Float32}, AbstractUnitRange{Int})
+        broken = (AbstractString,)
+        @testset "S: $S" for S ∈ stypes
+            @test test_expr((n -> :(function Base.show(::IO, ::$n) end)), S) broken=(S ∈ broken)
+        end
+    end
+    @testset "new subtype of `AbstractString`" begin
+        @testset "index-related" begin
+            fs = (thisind, prevind, nextind)
+            broken = (thisind, prevind, nextind)
+            @testset "f: $f" for f ∈ fs
+                @test test_expr((n -> :(function Base.$f(::$n, ::Int) end)), AbstractString) broken=(f ∈ broken)
+            end
+        end
+    end
+    @testset "new subtype of `Integer`" begin
+        int_types = (Bool, Int8, UInt8, Int16, UInt16, Int, UInt, BigInt)
+        @testset "construction of old int type from new type" begin
+            broken = (Int8, UInt8, UInt16, Int, UInt)
+            @testset "T: $T" for T ∈ int_types
+                @test test_expr((n -> :(function Base.$T(::$n) end)), Integer) broken=(T ∈ broken)
+            end
+        end
+        @testset "construction of new int type from old type" begin
+            broken = ()
+            @testset "T: $T" for T ∈ int_types
+                @test test_expr((n -> :(function $n(::$T) end)), Integer) broken=(T ∈ broken)
+            end
+        end
+        @testset "arithmetic between old int types and new int type" begin
+            ops = (:+, :*, :<<, :>>, :>>>)
+            broken = ()
+            @testset "T: $T" for T ∈ int_types
+                @testset "op: $op" for op ∈ ops
+                    @test (
+                        test_expr((n -> :(function Base.$op(::$n, ::$T) end)), Integer) &&
+                        test_expr((n -> :(function Base.$op(::$T, ::$n) end)), Integer)
+                    ) broken=((T, op) ∈ broken)
```

Comment:
This fails on 64-bit Windows :woozy_face:

What's worse is, there's no information on which test set failed specifically. Not sure how to diagnose more deeply.